lib/bundlex/nif.ex

```diff
@@ -1,131 +1,138 @@
-defmodule Bundlex.NIF do
+defmodule Bundlex.Native do
   alias Bundlex.{Helper, Output, Platform, Project}
   alias Helper.ErlangHelper
   use Bunch
 
   @type t :: %__MODULE__{
+          name: atom,
+          type: :nif | :cnode | :lib,
           includes: [String.t()],
           libs: [String.t()],
+          lib_dirs: [String.t()],
           pkg_configs: [String.t()],
           sources: [String.t()]
         }
-  defstruct includes: [], libs: [], pkg_configs: [], sources: []
 
-  def merge(%__MODULE__{} = nif1, %__MODULE__{} = nif2) do
-    Map.merge(nif1, nif2, fn
-      :__struct__, __MODULE__, __MODULE__ -> __MODULE__
-      _k, v1, v2 -> v1 ++ v2
-    end)
-  end
+  @enforce_keys [:name, :type]
+
+  defstruct name: nil,
+            type: nil,
+            includes: [],
+            libs: [],
+            lib_dirs: [],
+            pkg_configs: [],
+            sources: []
```

Comment:
There is an issue for adding arbitrary flags to the build. It doesn't seem like too much effort and can be solved here

Reply:
this will also be done in the next PR